### PR TITLE
Document equals and hashCode on Status

### LINKED ISSUE #751: Add javadoc to clarify equality on Status
Are Statuses equal based on code, code and message, or even as far as the throwable?  The javadoc for equals should clarify this.

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Resolves #751

### CODE REVIEW

File: core/src/main/java/io/grpc/Status.java [hunk 1]
```diff
@@ -459,4 +459,24 @@ public Status parseAsciiString(String serialized) {
...
+   * {@link #getCode}.  The description and cause of the Status are unlikely to be stable, and
+   * additional fields may be added to Status in the future.
+   */
+  @Override
+  public boolean equals(Object obj) {
+    return super.equals(obj);
+  }
+
+  /**
+   * Hash Codes on Statuses is not well defined.
```
Comment (nmittler) [inline core/src/main/java/io/grpc/Status.java]: nit: *are not well defined.

**Comment by nmittler** [comment]
LGTM

File: core/src/main/java/io/grpc/Status.java (hunk 1 above)
Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/Status.java]: The 'is' refers to equality which is singular.

**Comment by zhangkun83** [comment]
LGTM

**Comment by nmittler** [comment]
> Hash Codes on Statuses is not well defined.

>> The 'is' refers to equality which is singular.

"Equality" does not appear in that sentence.  The "is" refers to "Hash Codes", which is plural.  Not worth changing.

**Pull request closed without merging**